server/app/services/triplers.js

```diff
@@ -613,31 +613,24 @@ async function setTriplerEkataAssociatedPeople(tripler, verification) {
 
 async function syncTriplerHubSpot(tripler) {
   const email = tripler.get("alloy_person_id") + "@faux.blockpower.vote"
+  const query = "MATCH (t:Tripler {alloy_person_id: $alloy_person_id}) SET t.hs_id=toString($hs_id) RETURN t.id, t.hs_id"
     if(!tripler.get("hs_id")){
     await getContactHSID(email)
   }
 
   if (!tripler.get("hs_id")) {
     let obj = {}
-    console.log("no hs id, gettig it from hs")
+    console.log("[HS] This tripler has no hs_id, checking if contact already exists.")
     let hs_response = await getContactHSID(email)
-    if (!hs_response) {
-      obj["email"] = email
-      createHubspotContact(obj)
-      hs_response = await getContactHSID(email)
+    if (hs_response) {
+      let cypher_response = await neode.cypher(query,{alloy_person_id: tripler.get("alloy_person_id"),hs_id: hs_response})
     }
-
     if (!hs_response) {
-      return null
+    console.log("[HS] This tripler does not have an existing contact, creating a new one.")
+      obj["email"] = email
+      let hs_response = await createHubspotContact(obj)
```

Comment:
This doesn't look right — `createHubspotContact` returns the entire HTTP response body, not just the HubSpot ID.

In part this is confusing because up above the `hs_response` variable has a misleading name — the `hs_response` on line 624 should be called `hs_id` because it's just the ID, whereas this one could be rightfully called `hs_response` because it's the entire response body.

Reply:
cleaned up the names. createHubspotContact now returns the id.